src/protocol/ipa_prf/boolean_ops/comparison_and_subtraction_low_com.rs

```diff
@@ -0,0 +1,439 @@
+    let _ = subtraction_circuit(ctx, record_id, x, y, &mut carry).await;
+    Ok(carry)
+}
+
+///Comparison operation
+/// outputs x>y
+/// # Errors
+/// propagates errors from multiply
+#[cfg(all(test, unit_test))]
+pub async fn compare_gt<C, XS, YS>(
+    ctx: C,
+    record_id: RecordId,
+    x: &AdditiveShare<XS>,
+    y: &AdditiveShare<YS>,
+) -> Result<AdditiveShare<XS::Element>, Error>
+where
+    C: Context,
+    for<'a> &'a AdditiveShare<XS>: IntoIterator<Item = AdditiveShare<XS::Element>>,
+    YS: WeakSharedValue + CustomArray<Element = XS::Element>,
+    XS: WeakSharedValue + CustomArray + Field,
+    XS::Element: Field + std::ops::Not<Output = XS::Element>,
+{
+    let mut carry = AdditiveShare::<XS::Element>::ZERO;
+    // we don't care about the subtraction, we just want the carry
+    let _ = subtraction_circuit(ctx, record_id, x, y, &mut carry).await;
+    Ok(carry)
+}
+
+///non-saturated unsigned integer subtraction
+/// subtracts y from x, Output has same length as x (carries and indices of y too large for x are ignored)
+/// when y>x, it computes `(x+"XS::MaxValue")-y`
+/// # Errors
+/// propagates errors from multiply
+#[cfg(all(test, unit_test))]
+pub async fn integer_sub<C, XS, YS>(
+    ctx: C,
+    record_id: RecordId,
+    x: &AdditiveShare<XS>,
+    y: &AdditiveShare<YS>,
+) -> Result<AdditiveShare<XS>, Error>
+where
+    C: Context,
+    for<'a> &'a AdditiveShare<XS>: IntoIterator<Item = AdditiveShare<XS::Element>>,
+    YS: WeakSharedValue + CustomArray<Element = XS::Element>,
+    XS: WeakSharedValue + CustomArray + Field,
+    XS::Element: Field + std::ops::Not<Output = XS::Element>,
+{
+    let mut carry = AdditiveShare(XS::Element::ONE, XS::Element::ONE);
+    subtraction_circuit(ctx, record_id, x, y, &mut carry).await
+}
+
+///saturated unsigned integer subtraction
+/// subtracts y from x, Output has same length as x (we dont seem to need support for different length)
+/// when y>x, it outputs 0
+/// # Errors
+/// propagates errors from multiply
+#[cfg(all(test, unit_test))]
+pub async fn integer_sat_sub<C, S>(
+    ctx: C,
+    record_id: RecordId,
+    x: &AdditiveShare<S>,
+    y: &AdditiveShare<S>,
+) -> Result<AdditiveShare<S>, Error>
+where
+    C: Context,
+    for<'a> &'a AdditiveShare<S>: IntoIterator<Item = AdditiveShare<S::Element>>,
+    S: CustomArray + Field,
+    S::Element: Field + std::ops::Not<Output = S::Element>,
+{
+    let mut carry = AdditiveShare(S::Element::ONE, S::Element::ONE);
+    let result = subtraction_circuit(
+        ctx.narrow(&Step::SaturatedSubtraction),
+        record_id,
+        x,
+        y,
+        &mut carry,
+    )
+    .await?;
+
+    //carry computes carry=(x>=y)
+    //if carry==0 {all 0 array, i.e. Array[carry]} else {result}:
+    //compute (1-carry)*Array[carry]+carry*result =carry*result
+    let sat = AdditiveShare::<S>::expand(&carry)
+        .multiply(&result, ctx.narrow(&Step::MultiplyWithCarry), record_id)
+        .await?;
+    Ok(sat)
```

Comment:
Once again, while this is a really nice thing to have (everything is all zeroes when you go negative) this isn't required for the current OPRF circuit, and we can save a multiplication by skipping this.